### PR TITLE
add action for retrieving buttonData

### PR DESCRIPTION
**Pull request by mes1901**
https://trello.com/c/V900lC2R

### CODE REVIEW

File: frontend/src/services/visitors/averageSessionsService.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const fetchButtonValue = (startDate, endDate) => {
         'filter[startDate]': startDate,
         'filter[endDate]': endDate
     }).then(response => buttonTransformer(response.data))
-        .catch(err => throw err);
+        .catch(err => alert(err));
```
Comment (lenchvolodymyr) [inline frontend/src/services/visitors/averageSessionsService.js]: return back this

File: frontend/src/store/modules/visitors/actions.js [hunk 2]
```diff
@@ -9,11 +24,31 @@ export default {
...
     [CHANGE_FETCHED_BUTTON_STATE]: (context, data) => {
-
         if (data.value) {
             context.commit(SET_BUTTON_FETCHING, data.button);
         } else {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [GET_BUTTON_DATA]: (context, data) => {
+        if (data.value) {
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/visitors/actions.js]: do not wrap logic in block, it increses complexity of code. Better do

if (!data.value) {
    return;
}

File: frontend/src/store/modules/visitors/actions.js [hunk 3]
```diff
@@ -9,11 +24,31 @@ export default {
...
+            context.commit(SET_BUTTON_FETCHING, data.button);
+            context.commit(GET_SELECTED_PERIOD)
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/visitors/actions.js]: You can get period from context.state

File: frontend/src/store/modules/visitors/actions.js [hunk 4]
```diff
@@ -9,11 +24,31 @@ export default {
...
+                        .fetchButtonValue(response.startDate, response.endDate);
+                })
+                .then(response => {
+                    context.commit(SET_BUTTON_DATA, data.button, response.data);
+                })
+                .catch(err => {
+                    context.commit(RESET_BUTTON_FETCHING, data.button);
+                    throw err;
+                });
+            context.commit(RESET_BUTTON_FETCHING, data.button);
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/visitors/actions.js]: reset fetching after you got successful lresult. You can use .finally() of the promise

File: frontend/src/services/periodService.js [hunk 5]
```diff
@@ -38,7 +38,7 @@ const getWeekInterval = (week) => {
     return moment.duration({w: week}).asSeconds();
 };
 
-export const getTimeByPeriod = (period) => {
+const getTimeByPeriod = (period) => {
```
Comment (TatsuUkraine) [inline frontend/src/services/periodService.js]: Revert changes in this file

File: frontend/src/store/modules/visitors/actions.js [hunk 6]
```diff
@@ -9,11 +24,31 @@ export default {
...
+            context.commit(SET_BUTTON_FETCHING, data.button);
+            context.commit(GET_SELECTED_PERIOD)
+                .then(response => {
+                    periodService.getTimeByPeriod(response.data);
+                })
+                .then(response => {
+                    factoryVisitorService.create(data.button)
+                        .fetchButtonValue(response.startDate, response.endDate);
+                })
+                .then(response => {
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/visitors/actions.js]: you will never get this response, as you did not return previous value

File: frontend/src/store/modules/visitors/actions.js [hunk 7]
```diff
@@ -34,13 +39,39 @@ export default {
...
+                context.commit(SET_BUTTON_DATA, data.button, response.data);
+                context.commit(RESET_BUTTON_FETCHING, data.button);
+            })
+            .catch(err => {
+                context.commit(RESET_BUTTON_FETCHING, data.button);
+                throw err;
+            });
+
+    },
+    [GET_EACH_BUTTON_DATA]: (context) => {
```
Comment (TatsuUkraine) [inline frontend/src/store/modules/visitors/actions.js]: I'm not sure that you need this, you can simply get everything from state or with mapState or with getter

File: frontend/src/services/periodService.js [hunk 8]
```diff
@@ -38,11 +38,11 @@ const getWeekInterval = (week) => {
     return moment.duration({w: week}).asSeconds();
 };
 
-export const getTimeByPeriod = (value) => {
+const getTimeByPeriod = (period) => {
```
Comment (TatsuUkraine) [inline frontend/src/services/periodService.js]: Revert changes in this file

File: frontend/src/services/periodService.js [hunk 9]
```diff
@@ -38,11 +38,11 @@ const getWeekInterval = (week) => {
     return moment.duration({w: week}).asSeconds();
 };
 
-export const getTimeByPeriod = (value) => {
+export const getTimeByPeriod = (period) => {
     let interval = null;
     let startDate = null;
     let endDate = null;
-    switch (value) {
+    switch (period) {
```
Comment (lenchvolodymyr) [inline frontend/src/services/periodService.js]: If you don't changit it bake you break the logic period.PERIOD_TODAY

Comment (TatsuUkraine) [reply frontend/src/services/periodService.js]: I think changes in this file should be reverted entirely (see my last review)

**Pull request merged**